src/services/upper-bar-manager/models/icon-model.ts

```diff
@@ -14,8 +14,9 @@ export class IconModel {
   public componentRef: RefObject<IconWrapper>;
   public onClick: (e: MouseEvent | KeyboardEvent) => void;
   public component: ComponentClass<Record<string, never>> | FunctionalComponent<Record<string, never>>;
-  public svgIcon: SvgIcon;
+  public svgIcon: SvgIcon | (() => SvgIcon);
```

Comment:
why it should be function?

Reply:
the controls that I handled (PiP, CC, AAD, etc..) have 2 possible icons, which depends on the system's state. Hence, I added to the type a function that returns the SvgIcon.